modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java

```diff
@@ -0,0 +1,122 @@
+package com.ensolvers.fox.spring.lightweightcontainer;
+
+import com.ensolvers.fox.services.logging.Logger;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.List;
+import java.util.Vector;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.ApplicationContextException;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.stereotype.Component;
+
+
+/**
+ *      GlobalApplicationContext is a Spring companion.   <p>
+ *      Has two main features: look up and build.   <p>
+ *      <p>
+ *      <p>
+ *      * Look up allows to get from Spring context any registered bean  <p>
+ *      Example:  <code> MyComponent component = (MyComponent)GlobalApplicationContext.getBean( "MyComponent" );</code> <p>
+ *      <p>
+ *      <p>
+ *      * Construction allows building on the fly new context by offering legacy configuration xml files. <p>
+ *      Example:  <code> MyComponent component = (MyComponent)GlobalApplicationContext.getBean( "MyComponent" );</code> <p>
+ *
+ *      <p>
+ *
+ */
+@Component
+public class GlobalApplicationContext implements ApplicationContextAware {
+    private static ApplicationContext context;
+
+
+    public static void setGlobalContext( ApplicationContext globalContext ) {
+        Logger.info(GlobalApplicationContext.class, "setGlobalContext [" + globalContext + "]" );
+        context = globalContext;
+    }
+
+    public static Object getBean( String name ) {
+        if ( context == null ) {
+            Logger.error( GlobalApplicationContext.class, "No Spring Context available" );
+            throw new ApplicationContextException( "No Spring Context available" );
+        }
+        return getInstance().getBean(name);
+    }
+
+    public static ApplicationContext getInstance() {
+        return context;
+    }
+
+    public static Collection<String> getContextsFromProperty( InputStream stream ) throws IOException {
+        Vector contexts = new Vector();
```

Comment:
Vector is a very old class. The only perceivable advantage that I'm aware of is that the method is that their methods are synchronized. Of course that this comes with a cost. Do we need it? Can't we use an efficient implementation provided by  java.util.concurrent.* ?

Reply:
Now, re-reading this, why not using Apache IOUtils / FileUtils methods?

Reply:
Most important feature is that Vector has better and more usable protocol. Been synchronized or concurrent is of no use here, neither performance has an impact. 
Why use third party libraries if old proven Objects work well enough and are more versatile